src/inverseRobot.js

```diff
@@ -7,7 +7,23 @@
 */
 
 function inverseRobot(robot) {
-  // write code here
+  let newRobot = {};
+
+  const values = Object.values(robot);
+
+  for (let index = 0; index < values.length; index++) {
+    if (values.includes(values[index], index + 1)) {
+      newRobot = null;
+
+      return newRobot;
+    }
+  }
+
+  for (const key in robot) {
+    newRobot[robot[key]] = key;
+  }
```

Comment:
can you combine two loops into one?
hint: to check If any of the object values are repeated, you can check if robot[key] is already exist in newRobot in the secons loop

Reply:
Oh, really much better solution. Thank